app/Http/Controllers/UserController.php

```diff
@@ -822,12 +822,22 @@ public function postSignup()
         $lname = Input::get('lname');
         $user_details = Input::all();
 
+        $existingUser = User::where('email', $email)->first();
+        if (!empty($existingUser)) {
+            if (empty($existingUser->token)) {
+                return Response::json($this->growlMessage('The email address provided is already registered. You can <a href="/password/reset/">reset your password</a> if needed.', 'warning'), 400);
+            } else {
+                return $this->sendConfirmEmailWithResponse($existingUser);
+            }
+        }
+
         //Rules for signup form submission
-        $rules = array('email'        =>    'required|unique:users',
-                        'password'    =>    'required',
-                        'fname'        =>    'required',
-                        'lname'        =>    'required',
-                        );
+        $rules = array(
```

Comment:
I know this wasn't code written by you, but since you're in here it wouldn't be a bad idea to move this validation and authorization process to follow the [Form Request Validation](https://laravel.com/docs/5.1/validation#form-request-validation) pattern that Laravel provides. You can see a small example of [the object itself](https://github.com/opengovfoundation/madison/blob/dev/app/Http/Requests/UpdateDocumentRequest.php) and [using type hinting in the controller to apply it](https://github.com/opengovfoundation/madison/blob/dev/app/Http/Controllers/DocumentController.php#L161).

This is the direction we're trying to move for authorization and validation on requests.